hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSink.java

```diff
@@ -69,24 +69,25 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
 
       RowType rowType = (RowType) schema.toSinkRowDataType().notNull().getLogicalType();
 
+      // default parallelism
+      int parallelism = dataStream.getExecutionConfig().getParallelism();
+
```

Comment:
Can we just overwrite the parallelism of write/compaction/clustering tasks so that there is no need to pass around the default parallelism for each method.